src/app/@ansyn/plugins/openlayers/image-processing/image-processing.plugin.ts

```diff
@@ -1,59 +1,122 @@
-import { Observable, of } from 'rxjs';
+import { combineLatest, Observable, of } from 'rxjs';
 import ImageLayer from 'ol/layer/image';
-import Raster from 'ol/source/raster';
 import { BaseImageryPlugin, CommunicatorEntity, ImageryPlugin } from '@ansyn/imagery';
-import { ICaseMapState } from '@ansyn/core';
+import { ICaseMapState, ImageManualProcessArgs } from '@ansyn/core';
 import { IMapState, MapFacadeService, mapStateSelector } from '@ansyn/map-facade';
 import { Store } from '@ngrx/store';
 import { AutoSubscription } from 'auto-subscriptions';
 import { OpenLayersDisabledMap } from '../open-layers-map/openlayers-disabled-map/openlayers-disabled-map';
 import { OpenLayersMap } from '../open-layers-map/openlayers-map/openlayers-map';
 import { OpenLayersImageProcessing } from './image-processing';
-import { distinctUntilChanged, filter, map, tap } from 'rxjs/operators';
+import { distinctUntilChanged, filter, map, take, tap } from 'rxjs/operators';
+import { IImageProcParam, IToolsConfig, toolsConfig } from '@ansyn/menu-items';
+import { isEqual } from 'lodash';
+import { Inject } from '@angular/core';
 
 @ImageryPlugin({
 	supported: [OpenLayersMap, OpenLayersDisabledMap],
-	deps: [Store]
+	deps: [Store, toolsConfig]
 })
 export class ImageProcessingPlugin extends BaseImageryPlugin {
 	communicator: CommunicatorEntity;
 	private _imageProcessing: OpenLayersImageProcessing;
 	private imageLayer: ImageLayer;
+	customMainLayer = null;
+
 	currentMap$ = this.store$.select(mapStateSelector).pipe(
 		map((mapState: IMapState) => MapFacadeService.mapById(mapState.mapsList, this.mapId)),
 		filter(Boolean)
 	);
 
+	isAutoImageProcessingActive$ = this.currentMap$.pipe(
+		map((currentMap: ICaseMapState) => {
+			return currentMap.data.isAutoImageProcessingActive;
+		}),
+		distinctUntilChanged()
+	);
+
+	imageManualProcessArgsActive$ = this.currentMap$.pipe(
+		map((currentMap: ICaseMapState) => {
+			return currentMap.data.imageManualProcessArgs;
+		}),
+		distinctUntilChanged(isEqual)
+	);
+
 	@AutoSubscription
-	onToggleImageProcessing$: Observable<any> = this.currentMap$.pipe(
-		map((currentMap: ICaseMapState) => currentMap.data.isAutoImageProcessingActive),
-		distinctUntilChanged(),
-		filter(this.isImageLayerAndImageProcessing.bind(this)),
-		tap(this.setAutoImageProcessing.bind(this))
+	onAutoImageProcessingAndManualImageProcessActive$ = combineLatest(this.imageManualProcessArgsActive$, this.isAutoImageProcessingActive$).pipe(
+		filter(([imageManualProcessArgs, isAutoImageProcessingActive]: [ImageManualProcessArgs, boolean]) => {
+			return this.isImageProcessActive(isAutoImageProcessingActive, imageManualProcessArgs)
```

Comment:
change order  `this.isImageProcessActive(imageManualProcessArgs, isAutoImageProcessingActive)`
it's little bit confusing